### PR TITLE
Improve test coverage

### PR DESCRIPTION
**Pull request by bblommers**

### CODE REVIEW

File: moto/apigateway/exceptions.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ class NoIntegrationResponseDefined(NotFoundException):
 
     def __init__(self, code=None):
         super(NoIntegrationResponseDefined, self).__init__(
-            "NotFoundException", "No integration defined for method, code '%s'" % code
+            "NotFoundException", "Invalid Response status code specified"
```
Comment (bblommers) [inline moto/apigateway/exceptions.py:94]: This matches the official AWS response

File: moto/apigateway/exceptions.py [hunk 2]
```diff
@@ -161,7 +161,7 @@ class DomainNameNotFound(NotFoundException):
 
     def __init__(self):
         super(DomainNameNotFound, self).__init__(
-            "NotFoundException", "Invalid Domain Name specified"
+            "NotFoundException", "Invalid domain name identifier specified"
```
Comment (bblommers) [inline moto/apigateway/exceptions.py:164]: This matches the official AWS response

File: moto/apigateway/responses.py [hunk 3]
```diff
@@ -368,26 +369,21 @@ def stages(self, request, full_url, headers):
...
-        elif self.method == "DELETE":
-            self.backend.delete_stage(function_id, stage_name)
-            return 202, {}, "{}"
-        return 200, {}, json.dumps(stage_response)
+            elif self.method == "DELETE":
+                self.backend.delete_stage(function_id, stage_name)
+                return 202, {}, "{}"
+            return 200, {}, json.dumps(stage_response)
+        except StageNotFoundException as error:
+            return error.code, {}, error.get_body()
```
Comment (bblommers) [inline moto/apigateway/responses.py:386]: The models throw some exceptions, but they weren't properly handled yet, resulting in a wrong error coming from boto3

File: moto/awslambda/models.py [hunk 4]
```diff
@@ -1072,32 +1072,6 @@ def get_layer_version_by_arn(self, layer_version_arn):
         return None
 
 
-class LambdaPermission(BaseModel):
```
Comment (bblommers) [inline moto/awslambda/models.py:1075]: This class isn't used anywhere. The functionality is already handled in `class Permission` further down in this file

**Review by whummer — APPROVED** [review]
Looks great - really nice set of fixes and tests! 🚀

**Pull request merged**